util/file_util.cc

```diff
@@ -82,21 +82,32 @@ Status CreateFile(Env* env, const std::string& destination,
   return dest_writer->Append(Slice(contents));
 }
 
-Status DeleteSSTFile(const ImmutableDBOptions* db_options,
-                     const std::string& fname, uint32_t path_id) {
-  // TODO(tec): support sst_file_manager for multiple path_ids
 #ifndef ROCKSDB_LITE
-  auto sfm =
+Status DeleteDBFile(const ImmutableDBOptions* db_options,
+                    const std::string& fname, uint32_t path_id, FileType type) {
+  SstFileManagerImpl* sfm =
       static_cast<SstFileManagerImpl*>(db_options->sst_file_manager.get());
-  if (sfm && path_id == 0) {
+
+  if (sfm != nullptr &&
+      ((type == kTableFile && path_id == 0) || type == kLogFile)) {
```

Comment:
can you get rid of the `path_id == 0` restriction now that the trash files don't need to be moved?